### PR TITLE
feat(assistant): score URL validity and fix support ticket URL guidance

### PR DESCRIPTION
**Pull request by mattrossman**
**Logic changes**
- Adds function in `helpers.ts` to extract URLs from text via regex
  - I also considering using a library like [linkify-it](https://www.npmjs.com/package/linkify-it) for this but figured it's not worth the extra dep
- Adds associated tests in `helpers.test.ts`
- Adds "URL Validity" scorer which performs a HEAD request for links in Assistant response text and determins what portion of links have `.ok`  responses
- Adds eval case to check correctness of support ticket URL answers

**Prompt changes**
- Informs Assistant of https://supabase.com/dashboard/support/new being the URL to create support tickets 
- Encourages Assistant to "self-debug" issues before directing users to create support tickets

See [Eval Report](https://github.com/supabase/supabase/pull/42227#issuecomment-3807772871) and [Correctness](https://www.braintrust.dev/app/supabase.io/p/Assistant/trace?object_type=experiment&object_id=1ad0f9b0-5adb-436c-9812-a87aac62c036&r=1ef13459-a98c-4904-925e-6d81276cebb2&s=dbe5c607-a560-462b-8745-41d430744431) analysis for new support ticket test case.

Resolves AI-384

## Summary by CodeRabbit

* **New Features**
  * Added URL validity scoring to evaluations and helper utilities for extracting/cleaning URLs.
  * Added evaluation cases for support-ticket URL handling and OAuth callback guidance.

* **Documentation**
  * Updated assistant guidance to prefer self-resolution, include support-ticket direction, clarified data-recovery search steps, and added template-URL notation.

* **Tests**
  * Expanded URL extraction and related utility tests to cover many formats and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/studio/lib/helpers.test.ts [hunk 1]
```diff
@@ -311,6 +311,108 @@ describe('isValidHttpUrl', () => {
   })
 })
 
+describe.only('extractUrls', () => {
```
[the remark this answers is not in the document]
Comment (mattrossman) [reply apps/studio/lib/helpers.test.ts]: Resolved in https://github.com/supabase/supabase/pull/42227/commits/b5154c30561563c4149f02e3540207095b50695c

**Comment by mattrossman** [comment]
Sometimes Assistant writes template URLs that are representative of real URLs, e.g `https://project_ref.storage.supabase.co/storage/v1/s3`, which unintentionally hurts URL Validity score ([sample trace](https://www.braintrust.dev/app/supabase.io/p/Assistant/trace?object_type=experiment&object_id=e18a5f54-bdf7-405d-9674-571ca0d5c989&r=bdb90145-3fc1-4422-9518-db4dd0c593b7&s=129f7ad5-31e7-4ca1-ae81-90ba0ea90996)).

It also sometimes writes URLs in code snippet strings, which aren't meant to be interpreted as valid URLs all the time.

Two suggestions to work around these cases:
- Exclude URLs found in code blocks from scoring
- Instruct a more specific format for template URLs e.g. brackets like we demonstrate in docs [here](https://github.com/supabase/supabase/blob/4181b282a7a07227f3763f0680c636f741b51f2b/apps/docs/content/_partials/social_provider_setup.mdx#L1) and detect/exclude URLs with that syntax from the scorer

Edit: fixed in https://github.com/supabase/supabase/pull/42227/changes/c789407996c6a8eab2584042d59939608932a510

File: apps/studio/evals/scorer.ts [hunk 2]
```diff
@@ -323,3 +323,41 @@ export const sqlIdentifierQuotingScorer: EvalScorer<Input, Output, Expected> = a
...
+    urls,
+    errors: errors.length > 0 ? errors : undefined,
+  }
+
+  return {
+    name: 'URL Validity',
+    score: validUrls / urls.length,
+    metadata,
+  }
+}
```
[the remark this answers is not in the document]
Comment (mattrossman) [reply apps/studio/evals/scorer.ts:363]: Added timeout https://github.com/supabase/supabase/pull/42227/commits/d5ea9db7add81f53b9c80304a0286fe57908fd83

**Review by awaseem — APPROVED** [review]
LGTM
